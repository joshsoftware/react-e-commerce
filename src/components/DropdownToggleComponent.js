import React from 'react';
import { DropdownToggle } from 'reactstrap';
import ImageComponent from './ImageComponent';
import PropTypes from 'prop-types';

const DropdownToggleComponent = (props) => {
  const { src, className, alt } = props;

  return (
    <DropdownToggle nav caret>
      <ImageComponent style={{ width: '50px' }} src={src} className={className} alt={alt} />
    </DropdownToggle>
  );
};

export default DropdownToggleComponent;

DropdownToggleComponent.propTypes = {
  src: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired
};
