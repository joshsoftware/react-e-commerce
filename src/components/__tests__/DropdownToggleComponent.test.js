import DropdownToggleComponent from '../DropdownToggleComponent';
import { render } from '@testing-library/react';
import React from 'react';

describe('DropdowToggle Component', () => {
  it('Must return Dropdown Toggle', () => {
    const src =
      'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png';
    const className = 'rounded';
    const alt = 'img';
    const { asFragment } = render(
      <DropdownToggleComponent src={src} className={className} alt={alt} />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
