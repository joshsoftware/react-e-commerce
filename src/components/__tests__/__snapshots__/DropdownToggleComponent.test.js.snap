// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdowToggle Component Must return Dropdown Toggle 1`] = `
<DocumentFragment>
  <a
    aria-haspopup="true"
    class="dropdown-toggle nav-link"
    href="#"
  >
    <img
      alt="img"
      class="rounded"
      src="https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
      style="width: 50px;"
    />
  </a>
</DocumentFragment>
`;
